### DISCUSSION TITLE
Help needed configuring EKI1 / PSK vpn

### DISCUSSION THREAD

**Discussion by Sergio-Muriel** [strongswan/strongswan#893] [General]
Hello everybody,

I have a lot of troubles configuring the VPN for my work.
The vpn is based on an IKE auth and PSK pre-shared key.

The auth mecanism works fine: i am logged in  and i have an IP.
Problem: i am not able to ping anything and/or access anything in the network.

Here are the configuration information:

The swanctl.conf file is her: https://pastebin.com/umsD4bgT
_updown script is here: https://pastebin.com/gciAcs9N
Iptables rules: https://wtools.io/paste-code/b9wU

The ip addr's when connected:
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
2: enp3s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
    link/ether 18:31:bf:30:60:78 brd ff:ff:ff:ff:ff:ff
    inet 192.168.1.132/24 brd 192.168.1.255 scope global dynamic noprefixroute enp3s0
       valid_lft 81913sec preferred_lft 81913sec
**9: napali2@enp3s0: <NOARP,UP,LOWER_UP> mtu 1500 qdisc noqueue state UNKNOWN group default qlen 1000
    link/none 
    inet 172.25.12.23/32 scope global napali2
       valid_lft forever preferred_lft forever**
11: veth2674e53@if10: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br-2a0671084c37 state UP group default 
    link/ether 8e:ec:5f:dc:62:ec brd ff:ff:ff:ff:ff:ff link-netnsid 0
13: veth5c60b6f@if12: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br-2a0671084c37 state UP group default 
    link/ether ca:d3:f9:23:40:7c brd ff:ff:ff:ff:ff:ff link-netnsid 1
15: veth206c731@if14: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br-2a0671084c37 state UP group default 
    link/ether e2:09:e7:65:f8:49 brd ff:ff:ff:ff:ff:ff link-netnsid 2
17: veth8feb4cf@if16: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br-2a0671084c37 state UP group default 
    link/ether 0e:fb:97:e1:c7:82 brd ff:ff:ff:ff:ff:ff link-netnsid 3
18: ip_vti0@NONE: <NOARP> mtu 1480 qdisc noop state DOWN group default qlen 1000
    link/ipip 0.0.0.0 brd 0.0.0.0

Ip routes:
default via 192.168.1.1 dev enp3s0 proto dhcp metric 100 
172.17.0.0/16 dev docker0 proto kernel scope link src 172.17.0.1 linkdown 
172.18.0.0/16 dev br-2a0671084c37 proto kernel scope link src 172.18.0.1 
**172.24.0.0/16 dev napali2 scope link 
172.25.0.0/16 dev napali2 scope link** 
192.168.1.0/24 dev enp3s0 proto kernel scope link src 192.168.1.132 metric 100 

Result of ping:
⛔ root@eXistenZ:/home/tfe# ping 172.24.5.111
PING 172.24.5.111 (172.24.5.111) 56(84) bytes of data.
From 172.25.12.23 icmp_seq=1 Destination Host Unreachable
From 172.25.12.23 icmp_seq=2 Destination Host Unreachable

**Comment by tobiasbrunner** [comment]
Are you sure `remote_ts=dynamic` is what you want? You might want to try `remote_ts=0.0.0.0/0` to tunnel everything (or use the actual remote subnet), otherwise this will be narrowed to the peer's IP address.

Also, is there a specific reason for using an XFRM interface?

**Comment by Sergio-Muriel** [comment]
Thanks for the reply tobias.

Nop. I tried XFR because the "default" way was not working at all.

So here are the previous setup:


ipsec.conf:
config setup
   charondebug="all"
   uniqueids=yes

conn boardriders

    type = tunnel
    dpdaction = restart
    keyexchange = ikev1
    ike = aes128-sha1-modp1536,aes256-sha256-modp1536
    esp = aes128-sha1-modp1536,aes256-sha1-modp1536
    aggressive = yes
    ikelifetime=24h
    keylife=12h
    replay_window=32
    right = dns-of-the-server
    rightsubnet=0.0.0.0/0
    rightid = %any
    rightauth = psk

    leftfirewall = yes
    left = %defaultroute
    leftid = myuser
    leftsourceip = %config
    leftsubnet=%dynamic
    leftauth = psk
    leftauth2 = xauth
    xauth_identity = "myuser"
    #auto = route
    auto = start



⛔ root@eXistenZ:/home/tfe# ipsec  statusall
Status of IKE charon daemon (strongSwan 5.9.5, Linux 5.15.0-2-amd64, x86_64):
  uptime: 2 minutes, since Feb 11 17:45:59 2022
  malloc: sbrk 3973120, mmap 0, used 1974896, free 1998224
  worker threads: 11 of 16 idle, 5/0/0/0 working, job queue: 0/0/0/0, scheduled: 7
  loaded plugins: charon test-vectors ldap pkcs11 aesni aes rc2 sha2 sha1 md5 mgf1 rdrand random nonce x509 revocation pubkey pkcs1 pkcs7 pkcs8 pkcs12 pgp dnskey sshkey pem openssl gcrypt af-alg fips-prf gmp curve25519 agent chapoly xcbc cmac hmac ctr ccm gcm drbg curl attr kernel-netlink resolve socket-default connmark forecast farp stroke vici updown eap-identity eap-aka eap-md5 eap-gtc eap-mschapv2 eap-radius eap-tls eap-ttls eap-tnc xauth-generic xauth-eap xauth-pam tnc-tnccs dhcp lookip error-notify certexpire led addrblock unity counters
Listening IP addresses:
  192.168.1.132
  172.18.0.1
  172.17.0.1
Connections:
 boardriders:  %any...xxxxxxx  IKEv1 Aggressive, dpddelay=30s
 boardriders:   local:  [theuser] uses pre-shared key authentication
 boardriders:   local:  [theuser] uses XAuth authentication: any with XAuth identity 'theuser'
 boardriders:   remote: uses pre-shared key authentication
 boardriders:   child:  dynamic === 0.0.0.0/0 TUNNEL, dpdaction=restart
Security Associations (1 up, 0 connecting):
 boardriders[5]: ESTABLISHED 19 seconds ago, 192.168.1.132[theuser]...109.0.177.237[172.25.1.98]
 boardriders[5]: IKEv1 SPIs: 2e7bf7e6696fb668_i* 34612e0879cd4f84_r, pre-shared key+XAuth reauthentication in 23 hours
 boardriders[5]: IKE proposal: AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_1536
 boardriders{2}:  REKEYED, TUNNEL, reqid 1, expires in 11 hours
 boardriders{2}:   192.168.1.132/32 === 0.0.0.0/0
 boardriders{3}:  INSTALLED, TUNNEL, reqid 1, ESP in UDP SPIs: ce27694a_i 9f968cce_o
 boardriders{3}:  AES_CBC_128/HMAC_SHA1_96/MODP_1536, 0 bytes_i, 8094 bytes_o (66 pkts, 1s ago), rekeying in 11 hours
 boardriders{3}:   192.168.1.132/32 === 0.0.0.0/0


⛔ root@eXistenZ:/home/tfe# ip route show all
default via 192.168.1.1 dev enp3s0 proto dhcp metric 100 
172.17.0.0/16 dev docker0 proto kernel scope link src 172.17.0.1 linkdown 
172.18.0.0/16 dev br-2a0671084c37 proto kernel scope link src 172.18.0.1 
192.168.1.0/24 dev enp3s0 proto kernel scope link src 192.168.1.132 metric 100 

Same result: no ping to the DNS server (which should be accessible: tested from a working vpn access in windows)

With the configuration it looks that NO ip has been associated to my computer (i dont see anything in the ip addr result)

**Reply by tobiasbrunner** [reply]
Hm, your local traffic selector is `192.168.1.132/32`, which is your physical IP address. This is problematic as the remote host won't be able to route packets back to you. I don't know why that would be the case since you configured `leftsourceip = %config`, which should cause the request of a virtual IP address via mode config. (Check the log for messages regarding virtual IPs or mode config.)

**Comment by Sergio-Muriel** [comment]
Thanks Tobias,
I setup the log files and here are what i have: https://www.paste-txt.com/view/CFoVCVHtzK

I restarted the ipsec configuration (maybe i missed something in my previous test).
Now i have the following ipsec status:

Connections:
 boardriders:  %any...ipsec.thedns  IKEv1 Aggressive, dpddelay=30s
 boardriders:   local:  [theuser] uses pre-shared key authentication
 boardriders:   local:  [theuser] uses XAuth authentication: any with XAuth identity 'theuser'
 boardriders:   remote: uses pre-shared key authentication
 boardriders:   child:  dynamic === dynamic TUNNEL, dpdaction=restart
Security Associations (1 up, 0 connecting):
 boardriders[1]: ESTABLISHED 6 minutes ago, 192.168.1.132[theuser]...109.0.177.237[172.25.1.98]
 boardriders[1]: IKEv1 SPIs: 97e0ed54ad175b32_i* 533ef7428231d43a_r, pre-shared key+XAuth reauthentication in 23 hours
 boardriders[1]: IKE proposal: AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_1536
 boardriders{1}:  REKEYED, TUNNEL, reqid 1, expires in 11 hours
 boardriders{1}:   192.168.1.132/32 === 109.0.177.237/32
 boardriders{2}:  INSTALLED, TUNNEL, reqid 1, ESP in UDP SPIs: cbd72b99_i 9f968cf6_o
 boardriders{2}:  AES_CBC_128/HMAC_SHA1_96/MODP_1536, 0 bytes_i (0 pkts, 15s ago), 84 bytes_o (1 pkt, 130s ago), rekeying in 11 hours
 boardriders{2}:   192.168.1.132/32 === 109.0.177.237/32

It looks better, right?

but still no ping :(

Should i have a specific route for the private networks?
Should i have a specific iptables configuration?

**Comment by Sergio-Muriel** [comment]
So, i figured out that by putting:
` updown = /usr/lib/ipsec/_updown iptables`

i do get a virtual ip associated and iptables rules that looks almost good:
https://wtools.io/paste-code/b9xH

But when i run a ping:
⛔ root@eXistenZ:/home/tfe# ping 172.24.5.111
PING 172.24.5.111 (172.24.5.111) 56(84) bytes of data.
It is stuck.  It looks some routes are missing.

So, if i add a specific route for 172.24.0.0/16 network to the virtual ip:
$ ip route add 172.24.0.0/16 via 172.25.12.1

⛔ root@eXistenZ:/home/tfe# ping 172.24.6.86
PING 172.24.6.86 (172.24.6.86) 56(84) bytes of data.
From 172.25.12.1 icmp_seq=1 Destination Host Unreachable
From 172.25.12.1 icmp_seq=2 Destination Host Unreachable
From 172.25.12.1 icmp_seq=3 Destination Host Unreachable

I am missing something? :(

**Comment by Sergio-Muriel** [comment]
This was resolved by the IRC team.
